Tests/Services/NfieldSurveyGroupServiceTests.cs

```diff
@@ -0,0 +1,232 @@
+                .Returns(CreateTask(HttpStatusCode.OK));
+
+            var target = new NfieldSurveyGroupService();
+            target.InitializeNfieldConnection(mockedNfieldConnection.Object);
+
+            var result = await target.CreateAsync(new SurveyGroupValues
+            {
+                Name = "Default",
+                Description = null
+            });
+
+            Assert.NotNull(result);
+            Assert.Equal("Default", result.Name);
+            Assert.Equal(null, result.Description);
+            Assert.Equal(new DateTime(1799, 11, 10), result.CreationDate);
+            Assert.Equal(2, result.SurveyGroupId);
+
+            // modify survey group object we got back, then post it
+            result.Name = "Default with a twist";
+            result.Description = "Modified description";
+
+            result = await target.UpdateAsync(result.SurveyGroupId, result);
+
+            Assert.NotNull(result);
+            Assert.Equal("Default with a twist", result.Name);
+            Assert.Equal("Modified description", result.Description);
+            Assert.Equal(new DateTime(1799, 11, 10), result.CreationDate);
+            Assert.Equal(2, result.SurveyGroupId);
+
+            await target.DeleteAsync(result.SurveyGroupId);
+        }
+
+        [Fact]
+        public async Task CanGetAllSurveyGroups()
+        {
+            var expectedSurveyGroups = new[]
+            {
+                new SurveyGroup
+                {
+                    SurveyGroupId = 2,
+                    Name = "War & Peace",
+                    Description = "Napoleon satisfaction surveys",
+                    CreationDate = new DateTime(1812, 06, 26)
+                },
+                new SurveyGroup
+                {
+                    SurveyGroupId = 1,
+                    Name = "Default",
+                    CreationDate = new DateTime(1799, 11, 10)
+                }
+            };
+
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+            mockedHttpClient
+                .Setup(client => client.GetAsync(new Uri(ServiceAddress, "SurveyGroups")))
+                .Returns(CreateTask(HttpStatusCode.OK, new StringContent(JsonConvert.SerializeObject(expectedSurveyGroups))));
+
+            var target = new NfieldSurveyGroupService();
+            target.InitializeNfieldConnection(mockedNfieldConnection.Object);
+
+            var actualSurveyGroups = await target.GetAllAsync();
+
+            Assert.Equal(2, actualSurveyGroups.Count());
+
+            var defaultGroup = actualSurveyGroups.Single(sg => sg.SurveyGroupId == 1);
+            Assert.Equal("Default", defaultGroup.Name);
+            Assert.Equal(new DateTime(1799, 11, 10), defaultGroup.CreationDate);
+
+            var secondGroup = actualSurveyGroups.Single(sg => sg.SurveyGroupId == 2);
+            Assert.Equal("War & Peace", secondGroup.Name);
```

Comment:
Ditto
```suggestion
            Assert.Equal(expectedGroups.Where(g => g.SurveyGroupId == 2).Name ,  secondGroup.Name);
```

you do this similar logic using an array access instead of a LinQ expression, below

``` 
Assert.Equal(expectedSurveys[0].SurveyId, actualSurveys.ToArray()[0].SurveyId);
```

Check other asserts.